### ISSUE TITLE
Exporter in image does not respect ARCH

### ISSUE THREAD

**Issue by jensens** [mittwald/kube-httpcache#253]
**Describe the bug**

The metrics exporter fails with `exec /exporter/prometheus_varnish_exporter: exec format error`

**To Reproduce**
1. Enable exporter in `values.yaml`.
2. Apply Helm chart on a cluster with arm64 architecture

**Expected behavior**

A running metrics sidecar as I am used to on any Arch amd64 cluster

**Environment:**
 - Kubernetes version: 1.29
 - kube-httpcache version: v0.8.1

**Digging deeper:**

In 
https://github.com/mittwald/kube-httpcache/blob/2846c1f82c6c698a27338d47332cc82357066ebf/build/package/docker/GoReleaser.Dockerfile#L19
and 
https://github.com/mittwald/kube-httpcache/blob/2846c1f82c6c698a27338d47332cc82357066ebf/build/package/docker/GoReleaser.Dockerfile#L22

the architecture is not taken into account (while the ARG ARCH is present and used for the base image).

Unfortunately at https://github.com/jonnenauha/prometheus_varnish_exporter/releases there is not any arm64 package provided. 
I am going to create an issue there asking kindly for arm64 support.

**Comment by leontappe** [comment]
I'd prefer to actually have a arm64 Binary so we can respect the `ARCH` properly. It really should be no problem at all so I just forked the exporter to create a nice goreleaser Workflow that builds amd64 and arm64. Maybe this could be an easy drop in?

Successful builds are here https://github.com/leontappe/prometheus_varnish_exporter/releases/tag/v1.7.0-alpha.4

I will have to test this though, preferably less sleepy. I don't want to create more problems than solutions 😉

**Comment by pserrano** [comment]
@leontappe Yes, I think the same. For testing, I can run a test if you release a version in my test clusters. Happy to help if needed. That said, I believe it should work fine since it’s just a compilation change. Fortunately, thanks to Go, cross-compilation for different architectures is very straightforward.

**Comment by leontappe** [comment]
@pserrano I am building a pre-release right now, it would be great if you could test with that and report your findings

https://github.com/mittwald/kube-httpcache/releases/tag/v0.9.1-alpha.1

**Comment by leontappe** [comment]
@pserrano Just another heads-up: I made a v0.9.1 release with everything included in the alpha + some cleanup.

That should also resolve this issue in general, so I'll be closing it

**Comment by pserrano** [comment]
> [@pserrano](https://github.com/pserrano) Just another heads-up: I made a v0.9.1 release with everything included in the alpha + some cleanup.
> 
> That should also resolve this issue in general, so I'll be closing it

Hello @leontappe  looks like there are something wrong in the image: 

<img width="2013" height="43" alt="Image" src="https://github.com/user-attachments/assets/d85b2da9-cd03-4b6b-a90a-9281294f828b" />

**Comment by pserrano** [comment]
@leontappe also fails in amd64 as well
